imitation/scripts/data_collect.py

```diff
@@ -9,21 +11,112 @@
+      num_vec: Number of environments in VecEnv.
+      parallel: If True, then use DummyVecEnv. Otherwise use SubprocVecEnv.
+      make_blank_policy_kwargs: Kwargs for `make_blank_policy`.
+      rollout_save: Whether to save rollout files. If rollout_save is False,
+          then all the other `rollout_*` arguments are ignored.
+      rollout_save_interval: The number of training updates in between saves
+          after the first save. If the argument is `None`, then only save the
+          final update. Otherwise if the argument is an integer, then save
+          rollouts every `rollout_save_interval` updates and after the final
+          update.
+      rollout_save_n_samples: The number of timesteps saved in every file.
+      rollout_dir: The directory that rollouts are saved in.
+      policy_save: Whether to save policy files. If policy_save is False,
+          then all other `policy_*` arguments are ignored.
+      policy_save_interval: The number of training updates between saves. Has
+          the same semantics are `rollout_save_interval`.
+      policy_dir: The directory that policies are saved in.
+  """
+  with util.make_session():
+    tf.logging.set_verbosity(tf.logging.INFO)
+    sb_logger.configure(folder=osp.join(log_dir, 'rl'),
+                        format_strs=['tensorboard', 'stdout'])
+
+    env = util.make_vec_env(env_name, num_vec, seed=_seed,
+                            parallel=parallel, log_dir=log_dir)
+    # TODO(adam): add support for wrapping env with VecNormalize
+    # (This is non-trivial since we'd need to make sure it's also applied
+    # when the policy is re-loaded to generate rollouts.)
+    policy = util.make_blank_policy(env, verbose=1,
+                                    **make_blank_policy_kwargs)
+
+    callback = _make_callback(
+      env_name,
+      rollout_save, rollout_save_interval, rollout_save_n_samples,
+      rollout_dir, policy_save, policy_save_interval, policy_dir)
+
+    policy.learn(total_timesteps, callback=callback)
+
+    if rollout_save:
+      util.rollout.save_transitions(
+        rollout_dir, policy, env_name, "final", rollout_save_n_samples)
+    if policy_save:
+      util.save_policy(policy_dir, policy, env_name, "final")
+
+
+def _make_callback(env_name: str,
+                   rollout_save: bool = False,
+                   rollout_save_interval: Optional[int] = None,
+                   rollout_save_n_samples: Optional[int] = None,
+                   rollout_dir: Optional[str] = None,
+
+                   policy_save: bool = False,
+                   policy_save_interval: Optional[int] = None,
+                   policy_dir: Optional[str] = None,
+                   ) -> Callable:
+  """Make a callback that saves policy weights and rollouts during training.
+
+  Arguments are the same as arguments in `main()`.
+  """
+  step = 0
+  rollout_ok = rollout_save and rollout_save_interval is not None
```

Comment:
I think we should have one canonical way of enabling/disabling rollout/policy saving. So either mandate `rollout_save_interval` be an `int` (not `None`) and have `rollout_save` control it, or scrap `rollout_save` and make `rollout_save_interval` control it (`None` disabled, `int` enabled).

Right now I don't think `rollout_save_interval` should ever be `None`, as the Sacred config defaults to `50`.

Reply:
Right now, `rollout_save_interval is None` means final save only.
`rollout_save_interval is not None` means final save and intermediate saves.

I'll add `#` comments to `main` explaining this and then I'll check if Sacred becomes upset about switching config values between `NoneType` and `int`.

Reply:
We get a warning when changing `rollout_save_interval` from `int` type to `NoneType`.
```
(env) shwangster@aetherboard:~/repos/imitation$ python -m imitation.scripts.data_collect with rollout_save_interval=None
WARNING - root - Changed type of config entry "rollout_save_interval" from int to NoneType
```

I'll think a bit about if there's better ways to allow some combination of intermediate and final saving... Intuitively I think we can just do away with `rollout_save` and probably should make `rollout_save_interval` nonpositive instead of None.

Reply:
I've removed `rollout_save` and added with `rollout_final`. `rollout_save_interval`'s type was changed from `Optional[int]` to `int`. The new way to disable rollout saving now is to set `rollout_save_interval <= 0` and set to `rollout_final = False`.